### PR TITLE
Add cosmos container

### PR DESCRIPTION
**Pull request by jedheaj314**
# Resolves _#298_

## Unable to create cosmos container

Creating new cosmos container 

## Terraform changes

- Added resource block to add new cosmos container

### CODE REVIEW

File: apps/app/cosmos.tf [hunk 1]
```diff
@@ -34,3 +34,33 @@ resource "azurerm_cosmosdb_sql_role_assignment" "cosmos_access" {
   principal_id        = each.value
   scope               = "${data.azurerm_cosmosdb_account.state_store.id}/dbs/${azurerm_cosmosdb_sql_database.app.name}"
 }
+
+resource "azurerm_cosmosdb_sql_container" "example" {
+  name                  = "${var.app_id}-container"
+  resource_group_name   = var.resource_group_name
+  account_name          = var.cosmos_account_name
+  database_name         = azurerm_cosmosdb_sql_database.app.name
+  partition_key_path    = "/definition/id"
```
Comment (damoodamoo) [inline apps/app/cosmos.tf:43]: think should just be `/id`

File: apps/app/cosmos.tf [hunk 2]
```diff
@@ -34,3 +34,33 @@ resource "azurerm_cosmosdb_sql_role_assignment" "cosmos_access" {
...
+  partition_key_version = 1
+  throughput            = 400
```
Comment (damoodamoo) [inline apps/app/cosmos.tf:45]: remove throughput, don't need it here

File: apps/app/cosmos.tf [hunk 3]
```diff
@@ -34,3 +34,33 @@ resource "azurerm_cosmosdb_sql_role_assignment" "cosmos_access" {
...
+    included_path {
+      path = "/included/?"
+    }
+
+    excluded_path {
+      path = "/excluded/?"
+    }
+  }
+
+  unique_key {
```
Comment (damoodamoo) [inline apps/app/cosmos.tf:63]: take out `unique_key` - don't need it

File: apps/app/cosmos.tf [hunk 4]
```diff
@@ -34,3 +34,33 @@ resource "azurerm_cosmosdb_sql_role_assignment" "cosmos_access" {
   principal_id        = each.value
   scope               = "${data.azurerm_cosmosdb_account.state_store.id}/dbs/${azurerm_cosmosdb_sql_database.app.name}"
 }
+
+resource "azurerm_cosmosdb_sql_container" "example" {
+  name                  = "${var.app_id}-container"
```
Comment (damoodamoo) [inline apps/app/cosmos.tf:39]: let's take out `-container` , and just use the app_id as the name

**Review by damoodamoo — CHANGES_REQUESTED** [review]

File: apps/app/cosmos.tf (hunk 2 above)
Comment (jedheaj314) [reply apps/app/cosmos.tf]: Done ad608ff1cab05bf96cc5e42a0833e150468ebf03

File: apps/app/cosmos.tf (hunk 1 above)
Comment (jedheaj314) [reply apps/app/cosmos.tf]: Done ad608ff1cab05bf96cc5e42a0833e150468ebf03

**Review by damoodamoo — APPROVED** [review]

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4880727904 (with refid `5b112d31`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/299#issuecomment-1534303218) from @damoodamoo)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4880918189 (with refid `5b112d31`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/299#issuecomment-1534343537) from @jedheaj314)

File: apps/app/cosmos.tf [hunk 5]
```diff
@@ -34,3 +34,11 @@ resource "azurerm_cosmosdb_sql_role_assignment" "cosmos_access" {
   principal_id        = each.value
   scope               = "${data.azurerm_cosmosdb_account.state_store.id}/dbs/${azurerm_cosmosdb_sql_database.app.name}"
 }
+
+resource "azurerm_cosmosdb_sql_container" "example" {
```
Comment (tanya-borisova) [inline apps/app/cosmos.tf:38]: Do we want to name it `example`? 

What is this container for?

Comment (jedheaj314) [reply apps/app/cosmos.tf:38]: Creating container for whatever reason its needed. As devs/data scientist do no have permission to create one.
Renaming it to be just "container" instead as that sounds more appropriate.

Comment (jedheaj314) [reply apps/app/cosmos.tf]: Done f051821343519211d00dea34441b069ea912c6db

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4881554381 (with refid `5b112d31`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/299#issuecomment-1534468115) from @jedheaj314)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4881985943 (with refid `5b112d31`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/299#issuecomment-1534562029) from @jedheaj314)

**Pull request merged**